src/Renci.SshNet/Security/Cryptography/Cipher.cs

```diff
@@ -13,6 +13,14 @@ public abstract class Cipher
         /// </value>
         public abstract byte MinimumSize { get; }
 
+        /// <summary>
+        /// Gets the tag (MAC) size.
+        /// </summary>
+        /// <value>
+        /// The tag (MAC) size.
+        /// </value>
+        public virtual int TagSize { get; }
```

Comment:
Could it be done another way? It doesn't seem so good to put this on the base `Cipher` class

You had an `AeadCipher` class before. Putting this property there might be a bit better at least. Or perhaps replace `bool IsAead` with `int? AeadTagSize` on `CipherInfo` - null means it is not AEAD?

Reply:
[SSHJ](https://github.com/hierynomus/sshj) puts TagSize at BaseCipher. See https://github.com/hierynomus/sshj/blob/master/src/main/java/net/schmizz/sshj/transport/cipher/BaseCipher.java#L65

Using `AeadCipher` requires `as` operation in `Session` which seems not generic?
`int? AeadTagSize` implies AEAD when value is null which seems not intuitive?

Reply:
Could you please help me understand *it could even be enforced that the cipher is AeadCipher in KeyExchange.CreateCipher*

Reply:
Yes, sorry, I was thinking something like 

```c#
            // Create server cipher
            var cipher = _serverCipherInfo.Cipher(serverKey, serverVector);

            if (isAead && cipher is not AeadCipher)
            {
                throw new Exception("...");
            }

            return cipher;
```

but it is still not very elegant. You can leave it how it is